### ISSUE TITLE
Limitations?

### ISSUE THREAD

**Issue by yairchu** [LeviBorodenko/lancer#5]
It would be helpful for the readme to specify limitations.
For example I tried `lance` on the following example program and it broke it:

```Python
class Solution:
    def part1(self):
        print("AOC 1 Part 1")
    def part2(self):
        print("AOC 1 Part 2")

sol = Solution()
for i in [1, 2]:
    getattr(sol, f"part{i}")()
```

**Comment by LeviBorodenko** [comment]
Currently lancer is substituting comments and variable names. So ideally there really shouldn't be any limitations. Any that occur are bugs that need fixing!

**Comment by yairchu** [comment]
I think that to fully support the dynamic `getattr`s that Python allows would be almost impossible?

**Comment by LeviBorodenko** [comment]
Indeed. Now that I see the issue with the above example, I think that no static code analysis based approach will fix these. A limitation section in the readme would be appropriate.
